libimage/layer_tree.go

```diff
@@ -91,19 +91,38 @@ func (l *layerNode) repoTags() ([]string, error) {
 
 // layerTree extracts a layerTree from the layers in the local storage and
 // relates them to the specified images.
-func (r *Runtime) layerTree(ctx context.Context, images []*Image) (*layerTree, error) {
-	layers, err := r.store.Layers()
+// layerTree require both parameters not empty or nil, to avoid inconsistences
+// between layers and images. If both parameters are nil then layers and images will be obtained from local storage.
+// You can obtain consistent layer and images with storage.MultiList.
+func (r *Runtime) layerTree(images []*Image, layers []storage.Layer) (*layerTree, error) {
+	if (images == nil && layers != nil) || (images != nil && layers == nil) {
```

Comment:
Absolutely non-blocking: `if (images != nil) != (layers != nil)` is an option.

---

One perhaps elegant way to structure this is a switch:
```go
switch {
case /* inconsistent */: fail
case /* got data */: // nothing
default: // data misisng, load it
}